compiler-core/src/javascript/typescript.rs

```diff
@@ -90,11 +90,10 @@ fn collect_generic_usages<'a>(
 fn generic_ids(type_: &Type, ids: &mut HashMap<u64, u64>) {
```

Comment:
This generic_ids function is super hacky.

It would be better if the type information was "calculated" upstream and the typescript file would only print it.

The same type representation would be required if we are to compile to any type-checked language aside from typescript.

Reply:
What do you mean by hacky? Where would you like this calculated instead?